eva-remapping-ingest/src/test/java/uk/ac/ebi/eva/remapping/ingest/batch/io/RemappedSubmittedVariantsWriterTest.java

```diff
@@ -0,0 +1,194 @@
+        // Ensure the source variant we'd find is later than whatever is already in the db
+        SubmittedVariantEntity sourceSve = createSveInAssembly("GCA_000000001.5", 5000000002L, 2100, "C", "T",
+                                                               LocalDateTime.now(), null);
+        mongoTemplate.insert(sourceSve);
+
+        List<SubmittedVariantEntity> svesToWrite = Collections.singletonList(
+                createSve(5000000002L, 2100, "C", "T", LocalDateTime.now(), "GCA_000000001.5"));
+        writer.write(svesToWrite);
+        assertRemappingIngestCounts(0, 0, 1);
+    }
+
+    @Test
+    public void testDoesNotWriteNewDuplicateAccessions_discardFromDatabase() {
+        List<SubmittedVariantEntity> svesToWrite = Collections.singletonList(
+                createSve(5000000002L, 2100, "C", "T", LocalDateTime.now(), null));
+        writer.write(svesToWrite);
+        assertRemappingIngestCounts(1, 0, 1);
+    }
+
+    @Test
+    public void testDeduplicatesAccessionsFromInput() {
+        List<SubmittedVariantEntity> svesToWrite = Arrays.asList(
+                createSve(5000000003L, 2100, "C", "T", LocalDateTime.now(), "GCA_000000001.1"),
+                createSve(5000000003L, 2100, "A", "T", LocalDateTime.now(), "GCA_000000001.1"));
+        writer.write(svesToWrite);
+        assertRemappingIngestCounts(1, 0, 1);
+    }
+
+    @Test
+    public void testDiscardsDuplicateHashes() {
+        SubmittedVariantEntity sveWithSameHash = createSve(5000000002L, 2100, "C", "T", LocalDateTime.now(), null);
+        mongoTemplate.insert(sveWithSameHash);
+
+        List<SubmittedVariantEntity> svesToWrite = Collections.singletonList(
+                createSve(5000000005L, 2100, "C", "T", LocalDateTime.now(), "GCA_000000001.1"));
+        writer.write(svesToWrite);
+        assertRemappingIngestCounts(0, 0, 1);
+    }
+
+    @Test
+    public void testIdempotentWrites() {
+        List<SubmittedVariantEntity> svesToWrite = Collections.singletonList(
+                createSve(5000000002L, 2100, "C", "T", LocalDateTime.now(), null));
+        writer.write(svesToWrite);
+        assertRemappingIngestCounts(1, 0, 1);
+
+        Set<SubmittedVariantEntity> sveAfterFirstWrite = new HashSet<>(
+                mongoTemplate.findAll(SubmittedVariantEntity.class));
+        Set<SubmittedVariantOperationEntity> svoeAfterFirstWrite = new HashSet<>(
+                mongoTemplate.findAll(SubmittedVariantOperationEntity.class));
+
+        remappingIngestCounts.resetCounts();
+        writer.write(svesToWrite);
+        assertRemappingIngestCounts(0, 1, 0);
+
+        Set<SubmittedVariantEntity> sveAfterSecondWrite = new HashSet<>(
+                mongoTemplate.findAll(SubmittedVariantEntity.class));
+        Set<SubmittedVariantOperationEntity> svoeAfterSecondWrite = new HashSet<>(
+                mongoTemplate.findAll(SubmittedVariantOperationEntity.class));
+
+        assertEquals(sveAfterFirstWrite, sveAfterSecondWrite);
```

Comment:
You can also look for assertDatabaseState elsewhere in the codebase...

Reply:
`assertDatabaseState` in the clustering module is a bit overkill I think...  I'll leave this as is for now.